expression/builtin_string.go

```diff
@@ -2327,12 +2319,7 @@ func (b *builtinBitLengthSig) evalInt(row chunk.Row) (int64, bool, error) {
 	if isNull || err != nil {
 		return 0, isNull, err
 	}
-	argTp := b.args[0].GetType()
-	dBytes, err := charset.NewEncoding(argTp.Charset).Encode(nil, hack.Slice(val))
-	if err != nil {
-		return 0, isNull, err
-	}
-	return int64(len(dBytes) * 8), false, nil
+	return int64(len(val) * 8), false, nil
```

Comment:
Is it does not affect by the charset?

Reply:
It has been wrapped by `to_binary()`. The `val` here is the correct charset.